.vitepress/theme/components/ViteConfBanner.vue

```diff
@@ -0,0 +1,151 @@
+<script setup lang="ts">
+import ViteConfLogo from './ViteConfLogo.vue';
+import CloseIcon from './Icons/CloseIcon.vue';
+import { computed, ref, onMounted } from 'vue';
+
+const href =
+  'https://viteconf.org/?utm_campaign=stackblitz-on-page&utm_source=docs&utm_medium=nav-button';
+
+const key = 'shouldViteConfBannerBeShown';
+
+let shouldViteConfBannerBeShown = ref(false);
+
+onMounted(() => {
+  shouldViteConfBannerBeShown.value = window.localStorage.getItem(key) !== 'false';
+  if (shouldViteConfBannerBeShown.value) {
+    document.querySelector('.VPNav')?.classList.add('with-banner');
```

Comment:
This is super hacky way to manipulate Nav bar class, I'm so new to Vue/VitePress and I have zero idea how I can properly do this, if anyone knows I would like to know 👂

Reply:
I thought about controlling from CustomLayout but not sure how I can react changes in localstorage 🤔

Reply:
With Vitepress, not everything is necessarily as intuitive as it could be, and there's not a simple place to stick a "top level" component like, say, our Astro blog.

So if it works I think it's not a bad solution here 👍